### PR TITLE
Sets content type when the body is set if its presented.

### LINKED ISSUE #179: Java request API does not set Content-Type when setting the body
The Scala implementation of request sets the content type header when configuring the body for that request:

https://github.com/playframework/play-ws/blob/v1.0.2/play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala#L145-L159

The Java version just sets the body:

https://github.com/playframework/play-ws/blob/v1.0.2/play-ahc-ws-standalone/src/main/java/play/libs/ws/ahc/StandaloneAhcWSRequest.java#L249-L259

### PR DESCRIPTION
**Pull request by dmitraver**
# Pull Request Checklist

* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Typesafe CLA](https://www.typesafe.com/contribute/cla)?
* [x] Have you [squashed your commits](https://www.playframework.com/documentation/latest/WorkingWithGit#Squashing-commits)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

## Fixes

Fixes #179 

## Purpose

Sets content type header if its presented when the body is set.

### CODE REVIEW

File: play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala [hunk 1]
```diff
@@ -156,19 +167,19 @@ class AhcWSRequestSpec extends Specification with Mockito with DefaultBodyReadab
...
 
     "only send first content type header and keep the charset if it has been set manually with a charset" in {
       val client = mock[StandaloneAhcWSClient]
       val request = new StandaloneAhcWSRequest(client, "http://example.com", /*materializer*/ null)
-      request.setBody(body("HELLO WORLD"))
       request.addHeader("Content-Type", "application/json; charset=US-ASCII")
       request.addHeader("Content-Type", "application/xml")
+      request.setBody(body("HELLO WORLD")) // content type is not overwritten here as its already set before
       val req = request.buildRequest()
+      println("Headers:" + req.getHeaders)
```
Comment (marcospereira) [inline play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala]: Remove this. :-)

Comment (dmitraver) [reply play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala]: damnit, forgot to remove it, used for debugging :)

**Comment by dmitraver** [comment]
@marcospereira sure, I can add such test

**Comment by dmitraver** [comment]
@marcospereira done, I can squash the commits if needed.

**Comment by dmitraver** [comment]
@marcospereira hi, do you want to add something else to this change?

File: play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala [hunk 2]
```diff
@@ -152,22 +163,30 @@ class AhcWSRequestSpec extends Specification with Mockito with DefaultBodyReadab
...
+    "set content type explicitly if its not set with a body" in {
+      val client = mock[StandaloneAhcWSClient]
+      val request = new StandaloneAhcWSRequest(client, "http://example.com", /*materializer*/ null)
+      request.setBody(body("HELLO WORLD", null)) // content type is not set
+      request.addHeader("Content-Type", "application/json") // will be used as content type is not set with a body
+      val req = request.buildRequest()
+      req.getHeaders.get("Content-Type") must be_==("application/json")
+    }
+
     "only send first content type header and add charset=utf-8 to the Content-Type header if it's manually adding but lacking charset" in {
```
Comment (gmethvin) [inline play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala]: Can you update the description to match what you are testing here? It should be something like "not set Content-Type header if the body does not set it" and same pattern above "set Content-Type header if the body sets it".

File: play-ahc-ws-standalone/src/main/java/play/libs/ws/ahc/StandaloneAhcWSRequest.java [hunk 3]
```diff
@@ -248,13 +248,20 @@ public StandaloneAhcWSRequest setMethod(String method) {
 
     /**
      * Sets a BodyWritable directly.  See {@link DefaultBodyWritables} for common bodies.
+     * Also sets a Content-Type header if its presented in BodyWritable.
```
Comment (gmethvin) [inline play-ahc-ws-standalone/src/main/java/play/libs/ws/ahc/StandaloneAhcWSRequest.java]: "Also sets a Content-Type header if its specified in the BodyWritable and is not already set."

**Pull request merged**